src/Button.tsx

```diff
@@ -21,22 +23,35 @@ export interface DialogButtonProps extends TextProps {
   onPress: () => void;
 }
 
-const DialogButton: React.FC<DialogButtonProps> = (props) => {
+export interface DialogButtonHiddenProps {
+  vertical?: boolean;
+  buttonStyle?: ViewStyle;
+}
+
+const DialogButton: React.FC<DialogButtonProps & DialogButtonHiddenProps> = (
+  props
+) => {
   const {
     label,
     color = COLOR,
     disabled = false,
     bold,
     onPress,
     style,
+    buttonStyle = {},
+    vertical = false,
     ...nodeProps
   } = props;
   const fontWeight = bold ? "600" : "normal";
-  const { styles } = useTheme(buildStyles);
+  const buildStylesMemo = useMemo(
+    () => (isDark: boolean) => buildStyles(isDark, vertical),
```

Comment:
Good call making sure we keep the vertical/horizontal layout in sync with the props 👍 
Can we avoid updating the `buildStyles` signature for this though? I see that `StyleBuilder` in `useTheme` is now "aware" of the vertical params. Which I think causes two issues:
- I would prefer `StyleBuilder` to only handle "global" theme params, not component-specific ones. 
- Since these are not named parameters, if in the future we wanted to add another param to buildStyle it should be in the third sport, and so on...

Reply:
@mmazzarolo yes we can, I will add the extra styles needed and use ternary operators.